overlord/snapstate/autorefresh.go

```diff
@@ -974,3 +974,127 @@ func MockRefreshCandidate(snapSetup *SnapSetup) interface{} {
 		SnapSetup: *snapSetup,
 	}
 }
+
+func incrementSnapRefreshFailures(st *state.State, snapsup *SnapSetup) error {
+	var snapst SnapState
+	err := Get(st, snapsup.InstanceName(), &snapst)
+	if err != nil {
+		return err
+	}
+
+	// Update refresh failure information for snap revision
+	if snapst.RefreshFailures == nil {
+		// Initialize RefreshFailures
+		snapst.RefreshFailures = &snap.RefreshFailuresInfo{}
+	}
+	if snapst.RefreshFailures.ToRevision == snapsup.Revision() {
+		snapst.RefreshFailures.FailureCount++
+	} else {
+		snapst.RefreshFailures.ToRevision = snapsup.Revision()
+		snapst.RefreshFailures.FailureCount = 1
```

Comment:
we're starting from scratch, so let's use a new struct as well, so that you do not miss any fields::
```suggestion
		snapst.RefreshFailures = &snap.RefreshFailuresInfo{
			ToRevision: snapsup.Revision(),
			FailureCount: 1,
			LastFailureTime: timeNow(),
		}
```